crates/valence_registry/src/tags.rs

```diff
@@ -0,0 +1,95 @@
+use bevy_ecs::prelude::*;
+use bytes::Bytes;
+use serde::Deserialize;
+use valence_core::ident::Ident;
+use valence_core::protocol::encode::encode_packet;
+use valence_core::protocol::packet::synchronize_tags::*;
+
+#[derive(Debug, Resource)]
+pub struct TagsRegistry {
+    pub registries: Vec<Registry>,
+    cached_packet: Bytes,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Registry {
+    #[serde(rename = "registry")]
+    pub kind: Ident<String>,
+    pub tags: Vec<TagEntry>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct TagEntry {
+    pub name: Ident<String>,
+    #[serde(rename = "values")]
+    pub entries: Vec<u64>,
+}
```

Comment:
Would be nice to not need `serde(rename = ...)`. "kind" and "entries" were chosen arbitrarily.